src/AstRunner/Resolver/TypeResolver.php

```diff
@@ -145,6 +145,29 @@ public function resolveString(string $type, TypeScope $nameScope): array
         return $this->resolveReflectionType($resolvedType);
     }
 
+    /**
+     * @param \PhpParser\Node\Identifier|\PhpParser\Node\Name|\PhpParser\Node\NullableType|\PhpParser\Node\UnionType $type
+     *
+     * @return string[]
+     */
+    public function resolvePropertyType(NodeAbstract $type): array
+    {
+        if ($type instanceof Node\Name\FullyQualified) {
+            return [(string) $type];
+        }
+        if ($type instanceof Node\NullableType) {
+            return $this->resolvePropertyType($type->type);
+        }
+        if ($type instanceof Node\UnionType) {
+            return array_merge([], ...array_map(function (NodeAbstract $typeNode) {
+                /** @var \PhpParser\Node\Identifier|\PhpParser\Node\Name $typeNode */
```

Comment:
this line is introduced need to teach type to phpstan.

but, 
```
Files that were not fixed due to errors reported during linting before fixing:
   1) /home/runner/work/deptrac/deptrac/tests/AstRunner/Resolver/fixtures/PropertyTypeDependency.php
      ---------- begin diff ----------
--- Original
+++ New
@@ -160,7 +160,7 @@
         }
         if ($type instanceof Node\UnionType) {
             return array_merge([], ...array_map(function (NodeAbstract $typeNode) {
-                /** @var \PhpParser\Node\Identifier|\PhpParser\Node\Name $typeNode */
+                /* @var \PhpParser\Node\Identifier|\PhpParser\Node\Name $typeNode */
                 return $this->resolvePropertyType($typeNode);
             }, $type->types));
         }
```
https://github.com/sensiolabs-de/deptrac/pull/456/checks?check_run_id=1753259847

so, I would to change with `assert`
```php
                assert(
                    $typeNode instanceof \PhpParser\Node\Identifier ||
                    $typeNode instanceof \PhpParser\Node\Name
                );
```

Reply:
Can you try `@param` instead?

Reply:
likes this ?
```php
        if ($type instanceof Node\UnionType) {
            return array_merge([], ...array_map(
                /**
                 * @param \PhpParser\Node\Identifier|\PhpParser\Node\Name $typeNode
                 */
                function ($typeNode) {
                    return $this->resolvePropertyType($typeNode);
                },
            $type->types));
        }
```

Reply:
Yes

Reply:
changed with 4cb97b1

Reply:
Thanks